bgpd/bgp_vty.c

```diff
@@ -7387,6 +7387,35 @@ bgp_show_peer_afi (struct vty *vty, struct peer *p, afi_t afi, safi_t safi,
     }
 }
 
+static const char *
+text_sanitize (char *buf, size_t bufsz, const char *in, size_t inlen)
```

Comment:
https://tools.ietf.org/html/draft-ietf-idr-shutdown-06#section-2 - after the subcode, the length of the message follows. This length could be smaller than the size of the whole NOTIFICATION message.